dask/array/linalg.py

```diff
@@ -327,6 +343,8 @@ def tsqr(data, compute_svd=False, _max_vchunk_size=None):
             r_shape = (n, n) if data.shape[0] >= data.shape[1] else data.shape
             r_chunks = r_shape
 
+        dsk.dependencies[name_q_st3] = set(data.name)
```

Comment:
A couple more `set(data.name)` (which may not be a problem) 
https://github.com/dask/dask/pull/4081/files#r224559780